### ISSUE TITLE
Builds assume 100% availability of network services

### ISSUE THREAD

**Issue by tdcox** [jenkins-x/jx#856]
Using:

> NAME               VERSION
> jx                 1.2.63
> jenkins x platform 0.0.1027
> kubernetes cluster v1.9.7-gke.1
> kubectl            v1.9.6
> helm client        v2.9.0+gf6025bb
> helm server        v2.9.0+gf6025bb
> git                git version 2.15.1 (Apple Git-101)

I am seeing a lot of cases where builds are failing because network resources are transiently unavailable. This in turn leads to cluster status being left in indeterminante states. Suggest that the Circuit Breaker pattern be applied to all network connections to provide graceful degredation of retries in an inherently distributed environment.

For example, during a PR build, I have just seen the following:

```
helm repo add jenkins-x http://chartmuseum.build.cd.jenkins-x.io
Error: Looks like "http://chartmuseum.build.cd.jenkins-x.io" is not a valid chart repository or cannot be reached: Get http://chartmuseum.build.cd.jenkins-x.io/index.yaml: dial tcp 35.205.186.189:80: getsockopt: connection refused
```

**Comment by i0n** [comment]
Hi there! Our chartmuseum should be HA once https://github.com/jenkins-x/jx/pull/1430 is merged. I agree in terms of building in an expectation that network resources could be unavailable and mitigating that with retry/backoff or even circuit breaking if we can degrade gracefully. It's certainly something we are aware of and looking to improve as the project develops. I'm going to close this for now but please open a new issue if you'd like to discuss more or get involved in developing these features.

**Comment by abayer** [comment]
Reopening - we see this sort of issue pop up in BDD tests fairly frequently, so it's worth tackling for that on its own.

**Comment by tdcox** [comment]
Might be an idea to have a standard utility method for this, to make it super simple to implement?
